db/memorystore.go

```diff
@@ -132,11 +135,37 @@ func (m *memoryStore) addChallenge(chal *core.Challenge) (int, error) {
 	return len(m.challengesByID), nil
 }
 
-func (m *memoryStore) getChallengeByID(id string) *core.Challenge {
+func (m *MemoryStore) GetChallengeByID(id string) *core.Challenge {
 	m.RLock()
 	defer m.RUnlock()
 	if chal, present := m.challengesByID[id]; present {
 		return chal
 	}
 	return nil
 }
+
+func (m *MemoryStore) AddCertificate(cert *core.Certificate) (int, error) {
+	m.Lock()
+	defer m.Unlock()
+
+	certID := cert.ID
+	if len(certID) == 0 {
+		return 0, fmt.Errorf("cert must have a non-empty ID to add to MemoryStore")
+	}
+
+	if _, present := m.certificatesByID[certID]; present {
+		return 0, fmt.Errorf("cert %q already exists", certID)
+	}
+
+	m.certificatesByID[certID] = cert
+	return len(m.certificatesByID), nil
+}
+
+func (m *MemoryStore) GetCertificateByID(id string) *core.Certificate {
+	m.RLock()
+	defer m.RUnlock()
+	if cert, present := m.certificatesByID[id]; present {
```

Comment:
Good catch - this applies equally to the other memory store `GetX` functions. I'll simplify all of them with this round of review feedback.